lib/parse.js

```diff
@@ -335,7 +335,11 @@ function tokenizer($TEXT, filename, html5_comments, shebang) {
         if (prefix) num = prefix + num;
         var valid = parse_js_number(num);
         if (!isNaN(valid)) {
-            return token("num", valid);
+            var tok = token("num", valid);
+            if (num.indexOf('.') >= 0) {
+                tok.literal = num;
```

Comment:
Also note that integers can still be compacted in asm.js blocks:

    $ echo '"use asm"; f(1.00,2e0,300e5);' | bin/uglifyjs
    "use asm";f(1.00,2,3e7);